### PR TITLE
Reformat files

### PR DESCRIPTION
**Pull request by Chinay-Domitrix**
Reformatted the Rust, Markdown (and superscripted all the †s in the readme), SVG, and YAML files according to conventions; never touched Passerine files.

### CODE REVIEW

File: src/common/data.rs [hunk 1]
```diff
@@ -68,45 +60,46 @@ impl Display for Data {
...
         match self {
-            Data::Heaped(_)   => unreachable!("Can not display heaped data"),
-            Data::NotInit     => unreachable!("found uninitialized data on top of stack"),
-            Data::Real(n)     => write!(f, "{}", n),
-            Data::Integer(n)  => write!(f, "{}", n),
-            Data::Boolean(b)  => write!(f, "{}", if *b { "true" } else { "false" }),
-            Data::String(s)   => write!(f, "{}", s),
-            Data::Lambda(_)   => unreachable!("Can not display naked functions"),
-            Data::Closure(c)  => write!(f, "Function ~ {}", c.id),
-            Data::Kind(_)     => unreachable!("Can not display naked labels"),
```
Comment (slightknack) [inline src/common/data.rs:79]: Compilers tend to be sets of transformations on highly-structured tree-like data, which means that Passerine uses `match` extensively. Although I appreciate the consistency tools like `rustfmt` offer for enforcing code structure, they tend not to align match arms. Using `rustfmt` with Passerine is an ongoing discussion https://github.com/vrtbl/passerine/issues/40, but until https://github.com/rust-lang/rustfmt/issues/894 is resolved I value the improved readability of aligned match expressions over programmatically enforced stylistic consistency. Hope this makes sense!

File: src/common/lambda.rs [hunk 2]
```diff
@@ -86,7 +81,7 @@ impl Lambda {
             None => {
                 self.constants.push(data);
                 self.constants.len() - 1
-            },
```
Comment (slightknack) [inline src/common/lambda.rs:89]: I know I probably sounds like a broken record at this point, but another stylistic thing: I prefer trailing commas in match statements, even when used with a block.

File: src/common/lambda.rs [hunk 3]
```diff
@@ -95,7 +90,9 @@ impl Lambda {
         let mut best = &Span::empty();
 
         for (i, span) in self.spans.iter() {
-            if i > &index { break; }
```
Comment (slightknack) [inline src/common/lambda.rs:98]: I think single-line if-expressions should be placed on a single line.

File: src/common/lambda.rs [hunk 4]
```diff
@@ -143,66 +140,86 @@ impl fmt::Display for Lambda {
                 Opcode::Con => {
                     let (constant_index, consumed) = build_number(&self.code[index..]);
                     index += consumed;
-                    writeln!(f, "Load Con\t{}\t{:?}", constant_index, self.constants[constant_index])?;
```
Comment (slightknack) [inline src/common/lambda.rs:146]: Breaking this across multiple lines does not improve readability. This whole expression probably needs some cleanup, though.

File: src/common/number.rs [hunk 5]
```diff
@@ -11,8 +11,8 @@ pub fn split_number(n: usize) -> Vec<u8> {
     // high bit = last byte in chain?
     // low bits = binary representation
     let mut bytes = vec![];
-    let mut i     = n;
-    let chunk     = 0b1000_0000;
```
Comment (slightknack) [inline src/common/number.rs:15]: Multiple `let` statements ditto `match` arms.

File: src/common/number.rs [hunk 6]
```diff
@@ -49,9 +51,9 @@ pub fn build_number(bytes: &[u8]) -> (usize, usize) /* (index, eaten) */ {
         if byte >= &chunk {
             i += (byte - chunk) as usize;
             break;
-         } else {
-             i += *byte as usize;
-         }
```
Comment (slightknack) [inline src/common/number.rs:54]: Good catch!

File: src/common/source.rs [hunk 7]
```diff
@@ -15,7 +15,7 @@ use std::{
 #[derive(Debug, PartialEq, Eq)]
 pub struct Source {
     pub contents: String,
-    pub path:     PathBuf,
```
Comment (slightknack) [inline src/common/source.rs:18]: Ditto `struct` fields, `match` arms, and `let` expressions. What can I say, I like it when things align nicely ;P

File: src/common/span.rs [hunk 8]
```diff
@@ -59,9 +66,7 @@ impl Span {
     /// but false there is a total tie
     /// or otherwise.
     pub fn later_than(&self, other: &Span) -> bool {
-        self.offset > other.offset
-           || (self.offset == other.offset
-              && self.end() > other.end())
```
Comment (slightknack) [inline src/common/span.rs:64]: I think this is more readable, but it should probably be broken up into a couple expressions.

File: src/common/span.rs [hunk 9]
```diff
@@ -106,7 +115,9 @@ impl Span {
     /// so if the `Span` is along an invalid byte boundary or
     /// is empty, the program will panic.
     pub fn contents(&self) -> String {
-        if self.is_empty() { panic!("An empty span does not have any contents") }
+        if self.is_empty() {
+            panic!("An empty span does not have any contents")
+        }
```
Comment (slightknack) [inline src/common/span.rs:120]: I'm not a huge fan of breaking single expressions in blocks across multiple lines unless it really improves readability. I know readability is subjective; I think the amount of space source code occupies should be correlated with how difficult that code is to understand.

File: src/common/span.rs [hunk 10]
```diff
@@ -139,7 +150,13 @@ impl Debug for Span {
     // TODO: use the field, etc. constructor.
     fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
         if !self.is_empty() {
-            write!(f, "Span {{ {:?}, ({}, {}) }}", self.contents(), self.offset, self.length)
```
Comment (slightknack) [inline src/common/span.rs:142]: Same here: multiple lines doesn't add much. I know this was probably automatically formatted, but this is why I like to configure my automatic formatters.

**Review by slightknack — CHANGES_REQUESTED** [review]
Thank you so much for taking the time to format the codebase and fix some typos. I'd like to merge the changes to the Markdown, `.github/`, and `.svg`, but before I merge the mass changes to the Rust portion of the codebase, I think I should provide some context. I write this in a comment later down, but I'll move it here for ease-of-viewing:

> Compilers tend to be sets of transformations on highly-structured tree-like data, which means that Passerine uses `match` extensively. Although I appreciate the consistency tools like `rustfmt` offer for enforcing code structure, they tend not to align match arms. Using `rustfmt` with Passerine is an ongoing discussion https://github.com/vrtbl/passerine/issues/40, but until https://github.com/rust-lang/rustfmt/issues/894 is resolved I value the improved readability of aligned match expressions over programmatically enforced stylistic consistency.

Essentially, by default `rustfmt` mangles some organizational patterns I find essential for readability, and I'm waiting on an upstream issue before enforcing the use of the tool.

Other than that, looks great! There are a few minor stylistic changes in the README that need to be addressed, but other than removing/splitting out the Rust changes into a separate PR, I think it looks about ready to merge. Thanks you!

File: README.md [hunk 11]
```diff
@@ -45,25 +47,25 @@ Passerine is currently being developed by [Isaac Clayton](https://github.com/sli
...
-
-> ‡ Keith is currently [sponsoring](https://www.patreon.com/slightknack) the development of Passerine — I'm deeply grateful for his support!
+> <sup>†</sup> Shaw is writing an [alternative implementation of Passerine](https://github.com/ShawSumma/purr/tree/main/ext/passerine), and it's _super_ fast. It's part of a wider effort of his to develop [an efficient language-agnostic VM](https://github.com/ShawSumma/purr).<br/><sup>‡</sup> Keith is currently [sponsoring](https://www.patreon.com/slightknack) the development of Passerine — I'm deeply grateful for his support!
 
 ## An Overview
+
 Where this overview gets really exciting is when we dive into [macros](#macros). If you're here to give Passerine a try, [skip to Installation](#installation).
 
-> **⚠️ Note that Passerine is a *work in progress*: features mentioned in this overview may not be implemented yet.**
+> **⚠️ Note that Passerine is a _work in progress_: features mentioned in this overview may not be implemented yet.**
```
[the remark this answers is not in the document]
Comment (Chinay-Domitrix) [reply README.md:64]: That was Prettier, not me lol

File: ARCHITECTURE.md [hunk 12]
```diff
@@ -1,4 +1,5 @@
 # Passerine's Architecture
+
```
[the remark this answers is not in the document]
Comment (Chinay-Domitrix) [reply ARCHITECTURE.md:2]: Fair; the newline after headings is something enforced both by the Markdown error checker I use and by Prettier.

**Comment by Chinay-Domitrix** [comment]
As stated in [one of the reviews on the PR](https://github.com/vrtbl/passerine/pull/63#discussion_r745063869), most of the actual reformatting was done automatically (with either the official Rust extension for Open VSX or Prettier), and to be quite frank, I have little to no experience with Rust because of my background in Java and Kotlin, hence why all the manual changes are on the Markdown files and not the Rust files; I don't trust that I would be capable of making any Rust changes without fundamentally breaking something. In general, I'll likely undo my commit and then undo the Rust file changes before recommitting and pushing.

**Comment by slightknack** [comment]
No worries, and thank you for going through and correcting the markdown files by hand.

> In general, I'll likely undo my commit and then undo the Rust file changes before recommitting and pushing. I'll probably get around to it tomorrow.

Sounds like a plan :)

**Comment by Chinay-Domitrix** [comment]
I did it

**Comment by slightknack** [comment]
Let me take a look, 1 sec.

**Review by slightknack — CHANGES_REQUESTED** [review]
@Chinay-Domitrix I think there are still some changes to the Rust portion of the project, as well as some markdown changes (i.e. newlines after headings, italics, etc.) that need to be reverted. Can you double checked you've resolved and pushed everything? The number of files changed in this PR should be 4 or 5.

I think some changes may've been reverted when you merged `5a5d5e3` to produce `e22283e`?

**Comment by Chinay-Domitrix** [comment]
> @Chinay-Domitrix I think there are still some changes to the Rust portion of the project, as well as some markdown changes (i.e. newlines after headings, italics, etc.) that need to be reverted. Can you double checked you've resolved and pushed everything? The number of files changed in this PR should be 4 or 5.
> 
> I think some changes may've been reverted when you merged `5a5d5e3` to produce `e22283e`?

maybe I'll check

**Comment by Chinay-Domitrix** [comment]
Hmm im gonna jank this gimme a sec cause otherwise I have to manually undo everything

**Comment by slightknack** [comment]
> Hmm im gonna jank this gimme a sec cause otherwise I have to manually undo everything

Kk. If it's easier, you could probably just open another PR and close this one.

**Pull request closed without merging**